File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/domain/user/repository/UserRepository.java
package com.example.mediumspringwebflux.domain.user.repository;

import com.example.mediumspringwebflux.domain.user.document.User;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

public interface UserRepository extends ReactiveMongoRepository<User, String> {

    Mono<User> findByEmail(String email);
}

File: mediumSpringWebMVC/src/main/kotlin/com/example/mediumspringwebmvc/domain/user/repository/UserRepository.kt
package com.example.mediumspringwebmvc.domain.user.repository

import com.example.mediumspringwebmvc.domain.user.model.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Long> {

    fun findByEmail(email: String): User?
}